02.calendar/cal.rb

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env ruby
+
+require 'optparse'
+require 'date'
+
+opt = OptionParser.new
+opt.on('-m')
+opt.on('-y')
+
+opt.parse!(ARGV)
+month = ARGV[0] || Date.today.mon
+year = ARGV[1] || Date.today.year
+
+header = "#{month}月"+" "+year.to_s
+print header.center(20) + "\n"
+
+day_of_week = %w(日 月 火 水 木 金 土)
+  print " "+day_of_week.join(" ") + "\n"
+
+First_day = Date.new(year.to_i, month.to_i, 1)
+last_day = Date.new(year.to_i, month.to_i, -1)
+
+space = First_day.wday
+space.times do
+print "   "
+end
+
+days_of_the_month = (First_day..last_day)
+
+(days_of_the_month).each do |day|
+  number = day.wday
+  date_display = day.day.to_s.rjust(2)
+  case
+  when day == Date.today && number == 6
+    print " " + "\e[30;47m#{date_display}\e[0m" + "\n"
+  when day == Date.today
+    print " " + "\e[30;47m#{date_display}\e[0m"
+  when number == 6
+    print " " + date_display + "\n"
+  else
+    print " " + date_display
+  end
+end
```

Comment:
[IMO] 「今日だったら反転する」と「土曜日だったら改行する」という条件を
ひとつのケース式で分岐する必要はなく
別々の条件として考えてあげると、分岐がスッキリすると思います。

それから `Date#saturday?` というメソッドで土曜日かどうか判定できますので
それを使うのも良いと思います。

```suggestion
days_of_the_month.each do |day|
  date_display = day.day.to_s.rjust(2)
  if day == Date.today
    print " " + "\e[30;47m#{date_display}\e[0m"
  else
    print " " + date_display
  end
  print "\n" if day.saturday?
end
```

Reply:
@func09 
レビューありがとうございます！
以下、修正いたしました。お手数をおかけしますがご確認をお願いいたします🙇‍♀️

## 修正した点
### [MUST]
- yとmの引数の順番によって発生するエラーを解消
- First_dayの頭文字を小文字に（変数に）
  - 「月初日はどの年月でも1日である」から定数にしてしまっていましたが、改めてコードを確認すると`Date.new(year.to_i, month.to_i, 1)`のyearとmonthは引数で指定した数字が代入される=呼び出しごとに変わる、ことから変数が適切であると気づきました。

### [MUST]以外
-  年月の表示
  - 変数の使用をやめた
  - `print`→`puts`に変更
- 不要なインデントの削除
- 繰り返し処理を`n.times`から`String#*`に修正
- 「今日だったら反転する」と「土曜日だったら改行する」という条件を別々の条件とした分岐に修正
  - `Date#saturday?`メソッドを使用